packages/forma-36-react-components/src/components/Autocomplete/Autocomplete.tsx

```diff
@@ -0,0 +1,325 @@
+import React, { useReducer, useMemo, useRef, FunctionComponent } from 'react';
+import PropTypes from 'prop-types';
+import isHotKey from 'is-hotkey';
+
+import TextInput from '../TextInput';
+import Dropdown from '../Dropdown/Dropdown';
+import DropdownList from '../Dropdown/DropdownList';
+import DropdownListItem from '../Dropdown/DropdownListItem';
+import SkeletonBodyText from '../Skeleton/SkeletonBodyText';
+import SkeletonContainer from '../Skeleton/SkeletonContainer';
+import IconButton from '../IconButton';
+
+import styles from './Autocomplete.css';
+
+const TOGGLED_LIST = 'TOGGLED_LIST';
+const NAVIGATED_ITEMS = 'NAVIGATED_ITEMS';
+const QUERY_CHANGED = 'QUERY_CHANGED';
+const ITEM_SELECTED = 'ITEM_SELECTED';
+
+export interface AutocompleteProps {
+  children: (items: any[]) => any[];
+  items: any[];
+  onChange: (item: {}) => void;
+  onQueryChange: (query: string) => void;
+  disabled?: boolean;
+  placeholder?: string;
+  name?: string;
+  width?: 'small' | 'medium' | 'large' | 'full';
+  className?: string;
+  maxHeight?: number;
+  isLoading?: boolean;
+  emptyListMessage?: string;
+  noMatchesMessage?: string;
+  willClearQueryOnClose?: boolean;
+  isFullWidth?: boolean;
+}
+
+interface State {
+  isOpen?: boolean;
+  query: string;
+  highlightedItemIndex: number | null;
+}
+
+type Action =
+  | { type: typeof TOGGLED_LIST; payload?: boolean }
+  | { type: typeof NAVIGATED_ITEMS; payload: number | null }
+  | { type: typeof QUERY_CHANGED; payload: string }
+  | { type: typeof ITEM_SELECTED };
+
+enum Direction {
+  DOWN = 1,
+  UP = -1,
+}
+
+const initialState: State = {
+  isOpen: false,
+  query: '',
+  highlightedItemIndex: null,
+};
+
+const reducer = (state: State, action: Action): State => {
+  switch (action.type) {
+    case TOGGLED_LIST:
+      return {
+        ...state,
+        isOpen: action.payload,
+        highlightedItemIndex: null,
+      };
+    case NAVIGATED_ITEMS:
+      return {
+        ...state,
+        isOpen: true,
+        highlightedItemIndex: action.payload,
+      };
+    case QUERY_CHANGED:
+      return {
+        ...state,
+        highlightedItemIndex: null,
+        isOpen: true,
+        query: action.payload,
+      };
+    case ITEM_SELECTED:
+      return { ...initialState };
+    default:
+      return state;
+  }
+};
+
+export const Autocomplete: FunctionComponent<AutocompleteProps> = ({
```

Comment:
A named `export` and referencing `FunctionComponent` through it's named export, as opposed to `React.FunctionComponent` is necessary for `react-docgen-typescript-loader` to generate docs.